pdns/syncres.cc

```diff
@@ -2979,6 +2979,14 @@ void SyncRes::sanitizeRecords(const std::string& prefix, LWResult& lwr, const DN
       continue;
     }
 
+    if (rec->d_place == DNSResourceRecord::AUTHORITY && rec->d_type == QType::NS && !d_updatingRootNS && rec->d_name == g_rootdnsname) {
+      /* we don't want to pick up NS records in AUTHORITY and their ADDITIONALs sections of random queries
```

Comment:
Do you want to put the word 'root' in this comment, like in the PR title?

Also, the comment says AUTHORITY+ADDITIONAL but the code only says AUTHORITY?

Reply:
The `if (...AUTHORITY...)` checks determine what gets handled in the `if (...  ADDITIONAL ...)` block by manipulating `allowedAdditionals`. So while the code only checks for AUTHORITY it does impact how ADDITIONAL records are handled. I will change the comment to make it a bit more clear.